### PR TITLE
feature/component-library

### LINKED ISSUE #3: data-sources-app components → design system
### Context

We have some components [stacking up in our app](https://github.com/Police-Data-Accessibility-Project/data-sources-app/tree/main/client/src/components). Let's abstract these into reusable components in the design system, [ideally using Tailwind](https://github.com/Police-Data-Accessibility-Project/design-system/issues/9)!

### Requirements

This is a refactor, so things should work the same but details of appearance may change.

### Docs

If any best practices can be derived here for future front end developers, we should make changes to our internal docs or CONTRIBUTING.md or something. We can discuss this.

### Open questions

consider: should new css go in global.css or a use a component-level custom CSS file?
- for broadly used, lightly styled components, like containers, use `global-styles.css`
- for broadly used, uniquely styled components, like buttons, scope styles to the component. 
- for uniquely used, uniquely styled things like a map, scope styles to the component.
- Export `Vue.js` component library, or simply create the component styles here and reference the classes in the vue files elsewhere?

### PR DESCRIPTION
**Pull request by joshuagraber**
Resolves #3

### CODE REVIEW

**Comment by joshuagraber** [comment]
Hey @josh-chamberlain @mbodeantor @kalenluciano  Here's the start of the component library work, if you're interested in having a look. The config is complete and there is a lightweight `Button` component that works well in a `vue` project where I tested it locally.

My goals to complete the PR are as follows:
1. Add actual testing with `jest` and write one for the `Button` component.
2. Create lightweight versions of other components currently represented by PDAP UI and write associated tests.
3. Revise the `README` and `CONTRIBUTING` docs to ensure clarity for contributors to this repo and those consuming in other apps.

Once all of that is done, we should be g2g to both begin using these components/assets in projects and create a road map of work to continue building on this foundation.

**Comment by josh-chamberlain** [comment]
Thanks for the status update! Thanks for considering testing, too. Let us know if you want to sit down and check in about this, or how it relates to the app/site.

**Comment by kalenluciano** [comment]
This looks great! Like Josh said, let us know if you want to sit down and check in about anything

**Comment by joshuagraber** [comment]
Hey, @josh-chamberlain @kalenluciano @mbodeantor this is finally ready for review with the symlink strategy. (pull repo, `npm i` then `npm link` -> in test repo `npm link pdap-design-system`, then at root `import 'pdap-design-system/styles'`, then you should be good to `import { Header, Footer, etc } from 'pdap-design-system'` and use. 

Ignore the README for the time being. I'm working on a full set of documentation for these components, as well a test app to be contained within this repo to run the UI, but it's not quite ready. If you want to wait for that, it should be ready by tomorrow sometime. ✨

**Comment by josh-chamberlain** [comment]
@joshuagraber thank you! that sounds good. The symlinks are still good for testing changes to the design system on other apps, correct? or will there be a better way to do that, too?

File: src/components/Header/PdapHeader.vue
Comment (josh-chamberlain) [inline src/components/Header/PdapHeader.vue:1]: there are quite a few examples of class/component names starting with `Pdap`, any way we can ditch those?

Comment (joshuagraber) [reply src/components/Header/PdapHeader.vue:1]: Definitely can do that for the class names, but `Vue` is funny about having two-word component names, so for anything 1-word like `Button`, I've been defining as `PdapButton`, then in the enclosing directory's `index.ts` file, we do:

```
import PdapButton from './Pdapbutton.vue';

export { PdapButton as Button };
```

and then when we consume it, we can just do:
```
import { Button } from 'pdap-design-system';
```

For CSS class names, I personally like namespacing them with a prefix, but happy to revert if that's not desirable here.

**Comment by joshuagraber** [comment]
> The symlinks are still good for testing changes to the design system on other apps, correct? or will there be a better way to do that, too?

@josh-chamberlain Hey, yes the symlink is still the best strategy. I tried to nest a test app in the repo, but `Vue` and its router got confused. I think I need to do a little more research on how to isolate the sub-package within the same repo.

> I've had a little trouble getting things to work, because I'm in over my head without docs. I can wait!

Absolutely! Docs incoming, with detailed explanations of the components' APIs. Started today, will finish up Monday or Tuesday.

**Review by kalenluciano — APPROVED** [review]
This looks good to me! I'm not super familiar with setting up design systems, but this looks well-organized and makes sense to me reading through it.

**Pull request merged**